deployment/stacks/ui-auth.ts

```diff
@@ -0,0 +1,380 @@
+    | undefined = undefined;
+  let oktaIdp: cognito.CfnUserPoolIdentityProvider | undefined = undefined;
+
+  const providerName = "Okta";
+
+  if (oktaMetadataUrl) {
+    oktaIdp = new cognito.CfnUserPoolIdentityProvider(
+      scope,
+      "OktaUserPoolIdentityProviderSAML",
+      {
+        providerName,
+        providerType: "SAML",
+        userPoolId: userPool.userPoolId,
+        providerDetails: {
+          MetadataURL: oktaMetadataUrl,
+        },
+        attributeMapping: {
+          email:
+            "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress",
+          given_name:
+            "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname",
+          family_name:
+            "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname",
+          "custom:cms_roles": "cmsRoles",
+          "custom:cms_state": "state",
+        },
+        idpIdentifiers: ["IdpIdentifier"],
+      }
+    );
+    supportedIdentityProviders = [
+      cognito.UserPoolClientIdentityProvider.custom(providerName),
+    ];
+  } else if (oktaOidcClientId && oktaOidcClientSecret && oktaOidcIssuer) {
+    // TODO: This appears to never be used in any environment.
+    oktaIdp = new cognito.CfnUserPoolIdentityProvider(
+      scope,
+      "OktaUserPoolIdentityProviderOIDC",
+      {
+        providerName,
+        providerType: "OIDC",
+        userPoolId: userPool.userPoolId,
+        providerDetails: {
+          client_id: oktaOidcClientId,
+          client_secret: oktaOidcClientSecret,
+          oidc_issuer: oktaOidcIssuer,
+          attributes_request_method: "GET",
+          authorize_scopes: "email openid profile",
+        },
+        attributeMapping: {
+          email: "email",
+          given_name: "given_name",
+          family_name: "family_name",
+          "custom:cms_roles": "cms-roles",
+          "custom:cms_state": "state",
+        },
+        idpIdentifiers: ["IdpIdentifierOIDC"],
+      }
+    );
+    supportedIdentityProviders = [
+      cognito.UserPoolClientIdentityProvider.custom(providerName),
+    ];
+  }
+
+  const appUrl =
+    secureCloudfrontDomainName ||
+    applicationEndpointUrl ||
+    "http://localhost:3000/";
+
+  const userPoolClient = new cognito.UserPoolClient(scope, "UserPoolClient", {
+    userPoolClientName: `${stage}-user-pool-client`,
+    userPool,
+    authFlows: { adminUserPassword: true },
+    oAuth: {
+      flows: { authorizationCodeGrant: true },
+      scopes: [
+        cognito.OAuthScope.EMAIL,
+        cognito.OAuthScope.OPENID,
+        cognito.OAuthScope.PROFILE,
+      ],
+      callbackUrls: [appUrl],
+      logoutUrls: [appUrl, `${appUrl}postLogout`],
+    },
+    supportedIdentityProviders,
+    generateSecret: false,
+    accessTokenValidity: Duration.minutes(30),
+    idTokenValidity: Duration.minutes(30),
+    refreshTokenValidity: Duration.hours(24),
+  });
+
+  if (oktaIdp) {
+    userPoolClient.node.addDependency(oktaIdp);
+  }
```

Comment:
```suggestion
 userPoolClient.node.addDependency(oktaIdp);
```
Related to above suggestion.